### PR TITLE
JAMES-2224 Assert that the sender of a message is the owner before pu…

### PR DESCRIPTION
**Pull request by aduprat**
…tting it in outbox

### CODE REVIEW

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java [hunk 1]
```diff
@@ -2212,6 +2212,94 @@ public void setMessagesShouldRejectWhenSendingMessageHasMissingFrom() {
...
+        given()
+            .header("Authorization", accessToken.serialize())
+            .body(requestBody)
+        .when()
+            .post("/jmap")
+        .then()
+            .log().ifValidationFails()
+            .statusCode(200);
+
+        // Wait 10 seconds before checking that there's no message in Outbox
```
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java]: I don't understand why

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java [hunk 2]
```diff
@@ -251,6 +253,17 @@ private MessageWithId handleOutboxMessages(CreationMessageEntry entry, MailboxSe
         return new ValueWithId.MessageWithId(entry.getCreationId(), jmapMessage);
     }
 
+    private void assertUserIsSender(MailboxSession session, Optional<DraftEmailer> from) throws MailboxSendingNotAllowedException {
+        if (!from.map(DraftEmailer::getEmail)
+                .filter(Optional::isPresent)
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java]: `.flatMap(OptionalUtils::toStream)` (or something like that, I don't have my IDE opened ...)

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java [hunk 3]
```diff
@@ -251,6 +253,17 @@ private MessageWithId handleOutboxMessages(CreationMessageEntry entry, MailboxSe
...
+                .map(Optional::get)
+                .filter(email -> session.getUser().isSameUser(email))
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java]: anymatch

**Comment by aduprat** [comment]
test this please

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java [hunk 4]
```diff
@@ -251,6 +253,15 @@ private MessageWithId handleOutboxMessages(CreationMessageEntry entry, MailboxSe
         return new ValueWithId.MessageWithId(entry.getCreationId(), jmapMessage);
     }
 
+    private void assertUserIsSender(MailboxSession session, Optional<DraftEmailer> from) throws MailboxSendingNotAllowedException {
+        if (!from.flatMap(DraftEmailer::getEmail)
+                .filter(email -> session.getUser().isSameUser(email))
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java]: anymatch

Comment (aduprat) [reply server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java]: Not on Optional

**Pull request merged**